### PR TITLE
Testing branch

### PR DESCRIPTION
**Pull request by tonioshikanlu**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -13,11 +13,146 @@
...
+  	for (TimeRange optionalRange : optionalAvaliableTimes){
+  		for (TimeRange mandatoryRange : mandatoryAvaliableTimes){
+  			if (optionalRange.duration() >= request.getDuration()){
+  				if (mandatoryRange.contains(optionalRange) || mandatoryRange.overlaps(optionalRange)){
+  					allAvaliableTimes.add(mandatoryRange);
+  				}
+  			}
+  		}
+  	}
+  	/** Some edge cases
```
Comment (charu0291) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:48]: one liner comment?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -13,11 +13,146 @@
...
+				
+				if (lastRange.overlaps(range) || lastRange.contains(range)){
+					Integer newStart = Math.min(lastRange.start(), range.start());
+					Integer newEnd = Math.max(lastRange.end(), range.end());
+					stack.remove(stack.size() - 1);
+					stack.add(TimeRange.fromStartEnd(newStart, newEnd, false));
+				}
+					else {
+						stack.add(range);
+					}
```
Comment (charu0291) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:108]: fix identation

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -13,11 +13,146 @@
...
+  	}
+  	/** Some edge cases
+	*/
+	// Return times of only optional attendees if no mandatory attendees are indicated.
+  	if (mandatoryAttendees.isEmpty()){
+  		return optionalAvaliableTimes;
+  	}
+  	// Return times of only mandatory attendees if no optional attendees are indicated.
+  	if (optionalAvaliableTimes.isEmpty()){
+  		return mandatoryAvaliableTimes;
```
Comment (charu0291) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: check for edge case before for loop?

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 4]
```diff
@@ -223,6 +273,72 @@ public void justEnoughRoom() {
...
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList();
+
+    Assert.assertEquals(expected, actual);
+  }
+
+
   @Test
```
Comment (charu0291) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:342]: should we add test that include both optional and mandatory?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (tonioshikanlu) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:48]: resolved.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 4 above)
Comment (tonioshikanlu) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:342]: that test is on line 127: everyAttendeeIsConsideredOptionalNoChange()

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 5]
```diff
@@ -13,11 +13,144 @@
...
+  				}
+  			}
+  		}
+  	}
+  	// Return avaliable times for both mandatory and optional attendees.
+  	return allAvaliableTimes;
+  }
+  /** Helper function that determines the times the attendees passed into it are avaliable.
+  */
+  public Collection<TimeRange> avaliableTimesQuery(Collection<Event> events, MeetingRequest request, Collection<String> attendeesRequested) {
```
Comment (charu0291) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:61]: Break this function into multiple readable functions.

Comment (tonioshikanlu) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: resolved.
